src/riak_core_util.erl

```
@@ -262,6 +263,19 @@ build_tree(N, Nodes, Opts) ->
                     end, {[], tl(Expand)}, Nodes),
     orddict:from_list(Tree).
 
+%% Returns a forced-lowercase architecture for this node
+-spec get_arch () -> string().
+get_arch () -> string:to_lower(erlang:system_info(system_architecture)).
+
+%% Checks if this node is of a given architecture
+-spec is_arch (atom()) -> boolean().
+is_arch (linux) -> string:str(get_arch(),"linux") > 0;
+is_arch (darwin) -> string:str(get_arch(),"darwin") > 0;
+is_arch (sunos) -> string:str(get_arch(),"sunos") > 0;
+is_arch (osx) -> is_arch(darwin);
+is_arch (solaris) -> is_arch(sunos);
+is_arch (_) -> false.
```

Instead of just returning false I might throw here to make it clear it doesn't understand the passed in arch.